src/main/kotlin/org/dashevo/dpp/statetransition/StateTransitionFactory.kt

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2018-present, Dash Core Team
+ *
+ * This source code is licensed under the MIT license found in the
+ * COPYING file in the root directory of this source tree.
+ */
+
+package org.dashevo.dpp.statetransition
+
+import org.dashevo.dpp.contract.ContractFactory
+import org.dashevo.dpp.contract.ContractStateTransition
+import org.dashevo.dpp.document.Document
+import org.dashevo.dpp.document.DocumentsStateTransition
+import org.dashevo.dpp.errors.InvalidStateTransitionTypeError
+import org.dashevo.dpp.identity.IdentityCreateTransition
+import org.dashevo.dpp.util.Cbor
+
+class StateTransitionFactory() {
+
+    fun createStateTransition(rawStateTransition: MutableMap<String, Any?>, options: Options = Options()): StateTransition {
+        var stateTransition : StateTransition
+        when (StateTransition.Types.getByCode(rawStateTransition["type"] as Int)) {
+            StateTransition.Types.DATA_CONTRACT -> {
+                val dataContract = ContractFactory().createDataContract(rawStateTransition["dataContract"] as MutableMap<String, Any?>)
+
+                stateTransition = ContractStateTransition(dataContract);
+            }
+            StateTransition.Types.DOCUMENTS -> {
+                //this part will not work
```

Comment:
Can you please elaborate, maybe add a TODO?

Reply:
I forgot why this is there.  I will make a test for it and see if it actually doesn't work.

Reply:
It turns out that this code would not work because it was casting a List to a Map. A test was written to verify that this method works the same as other constructors.  #0909c11

Reply:
The comment was removed in this commit: c3babdd